### PR TITLE
WIP: Record protobuf

### PR DESCRIPTION
**Pull request by cube2222**

### CODE REVIEW

File: execution/distinct.go [hunk 1]
```diff
@@ -116,6 +120,7 @@ func (rs *recordSet) Insert(r *Record) (bool, error) {
 	return false, nil
 }
 
+//TODO: should the hash include other fields as well?
 func HashRecord(rec *Record) (uint64, error) {
-	return hashstructure.Hash(rec.data, nil)
+	return hashstructure.Hash(octosql.GetValuesFromPointers(rec.Data), nil)
```
Comment (cube2222) [inline execution/distinct.go]: Distinct shouldn't grab into internal fields.

File: execution/record.go [hunk 2]
```diff
@@ -197,3 +189,7 @@ type RecordStream interface {
 var ErrEndOfStream = errors.New("end of stream")
 
 var ErrNotFound = errors.New("not found")
+
+func NewMetadata() *Metadata {
```
Comment (cube2222) [inline execution/record.go]: Is this function necessary?

Comment (JasiekChomiak) [reply execution/record.go]: yeah it isn't - removing it

File: execution/distinct.go (hunk 1 above)
Comment (JasiekChomiak) [reply execution/distinct.go]: changed it so that Hash is now a method implemented on record (record.Hash)

**Pull request merged**